operator/internal/manifests/rules_config_test.go

```diff
@@ -24,15 +28,27 @@ func TestRulesConfigMap_ReturnsDataEntriesPerRule(t *testing.T) {
 
 func TestRulesConfigMap_ReturnsTenantMapPerRule(t *testing.T) {
 	opts := testOptions()
-	cm, err := manifests.RulesConfigMap(opts)
+	cm_shards, err := manifests.RulesConfigMapShards(opts)
 	require.NoError(t, err)
-	require.NotNil(t, cm)
+	require.NotNil(t, cm_shards)
+
+	cm := cm_shards[0]
 	require.Len(t, cm.Data, 4)
-	fmt.Print(opts.Tenants.Configs)
-	require.Contains(t, opts.Tenants.Configs["tenant-a"].RuleFiles, "dev-alerting-rules-alerts1.yaml")
-	require.Contains(t, opts.Tenants.Configs["tenant-a"].RuleFiles, "prod-alerting-rules-alerts2.yaml")
-	require.Contains(t, opts.Tenants.Configs["tenant-b"].RuleFiles, "dev-recording-rules-recs1.yaml")
-	require.Contains(t, opts.Tenants.Configs["tenant-b"].RuleFiles, "prod-recording-rules-recs2.yaml")
+	require.Contains(t, opts.Tenants.Configs["tenant-a"].RuleFiles, "-rules-0___tenant-a___dev-alerting-rules-alerts1.yaml")
+	require.Contains(t, opts.Tenants.Configs["tenant-a"].RuleFiles, "-rules-0___tenant-a___prod-alerting-rules-alerts2.yaml")
+	require.Contains(t, opts.Tenants.Configs["tenant-b"].RuleFiles, "-rules-0___tenant-b___dev-recording-rules-recs1.yaml")
+	require.Contains(t, opts.Tenants.Configs["tenant-b"].RuleFiles, "-rules-0___tenant-b___prod-recording-rules-recs2.yaml")
+}
+
+func TestRulesConfigMapSharding(t *testing.T) {
+	cm_shards, err := manifests.RulesConfigMapShards(testOptions_withSharding())
+	require.NoError(t, err)
+	require.NotNil(t, cm_shards)
+	require.Len(t, cm_shards, 2)
+}
+
+func TestDeleteRulesConfigMaps(t *testing.T) {
+
 }
```

Comment:
This is obsolete here right? The `cleanup.go` in the internal handlers package does this job, isn't it?

Reply:
Yup. `cleanup.go` handles everything